Lib/asyncio/base_events.py

```diff
@@ -1306,8 +1306,12 @@ def _check_sendfile_params(self, sock, file, offset, count):
 
             exceptions = []
 
-            if reuse_address is None:
-                reuse_address = os.name == 'posix' and sys.platform != 'cygwin'
+            if reuse_address:
```

Comment:
We want to deprecate and remove the parameter. Therefore:

* add somewhere in the file `_not_set = object()`; 
* change `reuse_address=None` to `reuse_address=_not_set` in the signature of the method;
* change this line to: `if reuse_address is not _not_set:`;
* handle two cases inside the `if`:  
   * `if reuse_address:` -- throw an error
   * `if not reuse_address:` -- issue a warning that the parameter is deprecated

Reply:
@1st1 Good point, the current version would only raise an error for `reuse_address=True`, but `reuse_address=False` should raise a deprecation warning since it's eventually going to be removed entirely. Is 3.11 okay for full removal of the parameter?

Reply:
@1st1 Done. I made a few minor changes from the above suggestions:

1) Renamed `_not_set` to `_unset`. This reads a bit better for the conditional check: `if reuse_address is not _unset` instead of `if reuse_address is not _not_set`. 

2) Replaced `if not reuse_address` with `else` since there are only two possible conditions to handle.